### PR TITLE
chore: update ajna contracts to `rc10` / `rc9.5`

### PR DESCRIPTION
**Pull request by halaprix**
## Ticket URL

Please provide a link to the ticket:

## Description of Changes

Please list the changes introduced by this PR:

- [ ] updated ajna contracts to rc10
- [ ] removed the usage of `balanceOf` to get withdrawn or repaid balance in favor of `_roundUpToScale` for repay and `_roundToScale` for withdraw
- [ ]

## How to Test

Please provide instructions on how to test the changes in this PR:

## PR Definition of Done

Please ensure the following requirements have been met before marking the PR as ready for review:

- [ ] All checks are passing
- [ ] PR is linked to a corresponding ticket
- [ ] PR title is clear and concise
- [ ] Code has been self-reviewed and any fixes or improvements noted (See Code review standards in Notion)
- [ ] Documentation has been updated if necessary

### CODE REVIEW

File: packages/ajna-contracts/scripts/common/config.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ export const TOKENS = {
     RETH: "0x0000000000000000000000000000000000000000",
     WSTETH: "0x0000000000000000000000000000000000000000",
     WBTC: "0x0000000000000000000000000000000000000000",
-    USDC: "0x6e4c6e76b3C1D834c0e3c4c2bAec8d58B8421A99",
+    USDC: "0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913",
```
Comment (robercano) [inline packages/ajna-contracts/scripts/common/config.ts:61]: Is this correct?

File: packages/ajna-contracts/scripts/common/deployment.utils.ts [hunk 2]
```diff
@@ -184,16 +184,19 @@ export async function deployPoolFactory(
...
 }
+
+// default rate = 4%
 export async function deployPool(
   erc20PoolFactory: ERC20PoolFactory,
   collateral: string,
   quote: string,
-  deployPools = true
+  deployPools = true,
+  rate = "40000000000000000"
```
Comment (robercano) [inline packages/ajna-contracts/scripts/common/deployment.utils.ts:194]: Rate value has changed from original implementation. Is this correct?

**Review by robercano — APPROVED** [review]
Couple of minor concerns, approved anyway

File: packages/ajna-contracts/scripts/common/deployment.utils.ts (hunk 2 above)
Comment (halaprix) [reply packages/ajna-contracts/scripts/common/deployment.utils.ts:194]: Yes, thats our ne default rate : https://discord.com/channels/837076147694207067/1150759720798388254/1180115026107957339

File: packages/ajna-contracts/scripts/common/config.ts (hunk 1 above)
Comment (halaprix) [reply packages/ajna-contracts/scripts/common/config.ts:61]: yes: https://basescan.org/token/0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913

**Pull request merged**